Sources/YPersistence/Manager/PersistenceManager+Sync.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  PersistenceManager+Sync.swift
+//  YPersistence
+//
+//  Created by Sahil Saini on 18/04/23.
+//  Copyright © 2023 Y Media Labs. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+extension PersistenceManager {
+    /// Fetches records from Core Data and convert those to models.
+    /// May be safely called from any thread, and the models returned may be safely transferred to any thread.
+    /// - Parameter context: Optional managed context to use. Default is `nil`.
+    /// - Returns: array of model objects not uploaded
+    /// - Throws: any error executing the fetch request
+    public func fetchRecordsToUpload<T: SyncRecord>(
+        context: NSManagedObjectContext? = nil
+    ) throws -> [T] {
+        let predicate = NSPredicate(format: "\(T.isUploadedKey) == false")
+        return try fetchRecords(
+            predicate: predicate,
+            sortDescriptors: T.sort?.descriptors
+        )
+    }
+
+    /// Fetches records from Core Data and convert those to models.
+    /// May be safely called from any thread, and the models returned may be safely transferred to any thread.
+    /// - Parameter context: Optional managed context to use. Default is `nil`.
+    /// - Returns: array of model objects to update
```

Comment:
```suggestion
    /// Fetches records that have been updated (but not deleted) since the last sync.
    ///
    /// May be safely called from any thread, but the managed objects returned
    /// may only be safely accessed from the current thread.
    /// If no context is provided, a worker context will be created.
    /// - Parameter context: Optional managed context to use. Default is `nil`.
    /// - Returns: an array of managed object records to update
```